opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1396,6 +1396,70 @@ namespace Opm
         }
     }
 
+    template<typename TypeTag>
+    bool
+    WellInterface<TypeTag>::
+    wellUnderZeroRateTargetVersion(const Simulator& ebos_simulator,
+                                   const WellState& well_state,
+                                   DeferredLogger& deferred_logger) const
+    {
+        // Extended version of WellInterfaceGeneric::wellUnderZeroRateTarget that also checks group controls
+        const auto& ws = well_state.well(this->index_of_well_);
+        const auto& summaryState = ebos_simulator.vanguard().summaryState();
+        const bool isGroupControlled = (this->isInjector() && ws.injection_cmode == Well::InjectorCMode::GRUP) ||
+                                       (this->isProducer() && ws.production_cmode == Well::ProducerCMode::GRUP);
+        if (!isGroupControlled) {
+            // well is not under group control, check "light-weight" version
+            return this->wellUnderZeroRateTarget(summaryState, well_state);
+        } else {
+            const auto& well = this->well_ecl_;
+            const auto& schedule = ebos_simulator.vanguard().schedule();
+            const auto& group_state = ebos_simulator.problem().wellModel().groupState();
+            const auto& group = schedule.getGroup(well.groupName(), this->currentStep());
+            const double efficiencyFactor = well.getEfficiencyFactor();
+            if (this->isInjector()) {
```

Comment:
@atgeirr , do you have some comment regarding how to determine the group rate target is zero?

Maybe we should not compared directly with `0.0` to consider the rounding error. 

There is more code in the group rate target determination, maybe they can go in a separate function.